src/platform/windows/misc.cpp

```diff
@@ -334,6 +401,33 @@ namespace platf {
     return true;
   }
 
+  /**
+
+@brief Check if the current process is running with system-level privileges.
+@return true if the current process has system-level privileges, false otherwise.
+*/
+  bool
+  is_running_as_system() {
+    // Retrieve the process token handle
+    HANDLE hToken;
+    OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &hToken);
+
+    PRIVILEGE_SET ps;
+    ps.PrivilegeCount = 1;
+    ps.Control = 0;  // check for any privilege
+    LookupPrivilegeValueW(NULL, L"SeDelegateSessionUserImpersonatePrivilege", &ps.Privilege[0].Luid);
```

Comment:
I'd prefer if we used an approach like `CreateWellKnownSid(WinLocalSystemSid)` and `CheckTokenMembership()`.